vowpalwabbit/learner.h

```diff
@@ -533,6 +537,14 @@ struct learner
   }
 };
 
+// OLD WAY:
+// The init_learner set of functions is the old way to create a learner. They
+// have been replaced with:
+//   - make_reduction_learner
+//   - make_base_learner
+// They were replaced due to how many different overloads there are and the
+// fact defaults are very hard to express. This problem got worse as more
+// arguments got added.
```

Comment:
do we want to deprecate?

Reply:
We can't deprecate as using a deprecated function is a build break in the warning as error CIs

Reply:
True just saw that only the print reduction has been migrated